projects/hslayers/common/layman/access-rights/layman-access-rights.component.ts

```diff
@@ -91,105 +170,257 @@ export class HsCommonLaymanAccessRightsComponent implements OnInit {
+  ): boolean {
+    //Switching from users table
+    if (this.currentOption === GrantingOptions.PERUSER) {
+      //Users with the role
+      const usersWithRole = this.allUsers.filter((u) => u.role.includes(role));
+      //Users of that role with access
+      const usersWithRoleAndAccess = usersWithRole.filter((u) => u[type]);
+      return (
+        //True in case more than half of the users from the role has access
+        (usersWithRoleAndAccess.length !== 0 &&
+          usersWithRoleAndAccess.length >= usersWithRole.length / 2) ||
+        //Or all users has access
+        this.allUsers.filter((u) => u[type]).length === this.allUsers.length
       );
     }
+    //Switching from 'everyone' tab or opening role tab first
+    return rights.includes(role) || rights.includes('EVERYONE');
   }
 
   /**
-   * Change access granting options (everyone or per_user)
-   * @param option - Access granting option
+   * Get user roles
    */
-  async changeGrantingOptions(option: 'per_user' | 'everyone'): Promise<void> {
-    if (option == this.grantingOptions.PERUSER) {
-      await this.getAllUsers();
-      this.access_rights['access_rights.read'] = this.endpoint.user;
-      this.access_rights['access_rights.write'] = this.endpoint.user;
-    } else {
-      this.allUsers = [];
-      this.access_rights['access_rights.read'] = 'EVERYONE';
-      this.access_rights['access_rights.write'] = 'private';
+  async getRoles(access_rights?: AccessRightsModel): Promise<void> {
+    if (this.endpoint?.authenticated) {
+      const url = `${this.endpoint.url}/rest/roles`;
+
+      access_rights ??= this.access_rights;
+      const read = access_rights[AccessRights.READ].split(',');
+      const write = access_rights[AccessRights.WRITE].split(',');
+
+      this.allRoles = await lastValueFrom(
+        of(this.allRoles).pipe(
+          switchMap((roles) =>
+            roles.length === 0
+              ? this.$http.get<string[]>(url, {withCredentials: true}).pipe(
+                  catchError((error) => {
+                    console.warn('Could not get roles  list', error);
+                    return of([]);
+                  }),
+                )
+              : of(roles.map((r) => r.name)),
+          ),
+          map((roles: string[]) => {
+            return roles
+              .filter((r) => r !== 'EVERYONE')
+              .map((r) => {
+                return {
+                  name: r,
+                  read: this.roleHasAccess(r, read, 'read'),
+                  write: this.roleHasAccess(r, write, 'write'),
+                };
+              });
+          }),
+        ),
+      );
+      this.setAcessRightsFromActor('read', GrantingOptions.PERROLE);
+      this.setAcessRightsFromActor('write', GrantingOptions.PERROLE);
     }
-    this.currentOption = option;
+  }
+
+  /**
+   * Check whether user list includes wagtail user list
+   */
+  private isWagtailUserArray(users: any[]): users is WagtailUser[] {
```

Comment:
Wow, new syntax to me 👀

Reply:
Type guard kinda nice feature:
https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates

Thanks for pointing this out. That any can and should be replaced with users: LaymanUser[] | WagtailUser[].